File: frontend/src/app/features/dashboard/partner/partner-membership-dto.ts
import {PartnerMembershipRole} from '../../../shared/partner-membership-role';

export interface PartnerMembershipDto {
  id?: number;
  userId: number;
  userName: string;
  email: string;
  phone: string;
  partnerId: number;
  partnerName: string;
  role: PartnerMembershipRole;
}

File: backend/src/main/java/hu/ibc/ibcpartners/partner/dto/PartnerMembershipDto.java
package hu.ibc.ibcpartners.partner.dto;

import hu.ibc.ibcpartners.partner.entity.PartnerMembershipRole;

public record PartnerMembershipDto(
        Long id,
        Long userId,
        String userName,
        String email,
        String phone,
        Long partnerId,
        String partnerName,
        PartnerMembershipRole role) {
}
